### PR TITLE
Add surface_dynamics

### PR DESCRIPTION
**Pull request by saraedum**
* [x] ~~The license is not packaged yet https://gitlab.com/videlec/surface_dynamics/merge_requests/5~~

### CODE REVIEW

**Comment by saraedum** [comment]
@videlec @slel would you mind if I add you as maintainers for this package?

**Comment by slel** [comment]
By all means do.

**Comment by videlec** [comment]
There is now a 0.4.3 version on PyPI shipping a license file.

You can put me as maintainer.

**Comment by saraedum** [comment]
@conda-forge/help-python This is ready for review :)
@isuruf: Since this is SageMath related, you might want to have a look.

File: recipes/surface-dynamics/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - sagelib
+    # LattE Integrale is recommended, but it is not packaged yet: https://github.com/conda-forge/staged-recipes/pull/8959
+
+test:
+  imports:
+    - surface_dynamics
+
+about:
+  home: http://www.labri.fr/perso/vdelecro/surface-dynamics/latest/
+  license: GPLv2+
```
Comment (chrisburr) [inline recipes/surface-dynamics/meta.yaml]: We prefer to use the SPDX identifiers for consistency:
```suggestion
  license: GPL-2.0-or-later
```

File: recipes/surface-dynamics/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    # surface_dynamics.interval_exchanges.iet_family uses C++
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - pip
+    - sagelib
+  run:
+    - python
+    - sagelib
+    # LattE Integrale is recommended, but it is not packaged yet: https://github.com/conda-forge/staged-recipes/pull/8959
```
Comment (chrisburr) [inline recipes/surface-dynamics/meta.yaml]: Is it worth waiting for https://github.com/conda-forge/staged-recipes/pull/8959 to be fixed and merged?
```suggestion
    - latte-integrale
```

**Comment by isuruf** [comment]
```
WARNING (surface-dynamics,lib/python2.7/site-packages/surface_dynamics/interval_exchanges/iet_family.so): Needed DSO lib/libgmp.so.10 found in ['gmp']
WARNING (surface-dynamics,lib/python2.7/site-packages/surface_dynamics/interval_exchanges/iet_family.so): .. but ['gmp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (surface-dynamics,lib/python2.7/site-packages/surface_dynamics/interval_exchanges/iet_family.so): Needed DSO lib/libppl.so.14 found in ['ppl']
WARNING (surface-dynamics,lib/python2.7/site-packages/surface_dynamics/interval_exchanges/iet_family.so): .. but ['ppl'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

File: recipes/surface-dynamics/meta.yaml (hunk 1 above)
Comment (saraedum) [reply recipes/surface-dynamics/meta.yaml]: Thanks I didn't know :)

**Comment by saraedum** [comment]
> ```
> WARNING (surface-dynamics,lib/python2.7/site-packages/surface_dynamics/interval_exchanges/iet_family.so): Needed DSO lib/libgmp.so.10 found in ['gmp']
> WARNING (surface-dynamics,lib/python2.7/site-packages/surface_dynamics/interval_exchanges/iet_family.so): .. but ['gmp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
> WARNING (surface-dynamics,lib/python2.7/site-packages/surface_dynamics/interval_exchanges/iet_family.so): Needed DSO lib/libppl.so.14 found in ['ppl']
> WARNING (surface-dynamics,lib/python2.7/site-packages/surface_dynamics/interval_exchanges/iet_family.so): .. but ['ppl'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
> ```

Thanks, should be fixed now.

File: recipes/surface-dynamics/meta.yaml (hunk 2 above)
Comment (saraedum) [reply recipes/surface-dynamics/meta.yaml]: From looking at the source it only adds a few features but is not essential.

**Pull request merged**